### PR TITLE
Allow auth hostnames other than arxiv.org

### PR DESCRIPTION
**Pull request by bdc34**
The goal is serve pages to both arxiv.org and web7.arxiv.org so that when VMs are being setup individual VMs can be tested.

The flask framework blueprints will switch into a subdomain mode when `SERVER_NAME` is set. This will cause blueprints to only route when the Host matches the `SERVER_NAME` or the hostname set on the blueprint. 

In the past `SERVER_NAME` was set to arxiv.org and this worked for production. But it caused https://web7.arxiv.org/login to return a 404 since `web7.arxiv.org` did not match the value in `SERVER_NAME`.

So this forces the flask app.config `SERVER_NAME` to be None.

This is running on web7 now and web7.arxiv.org is a DNS name and the httpd.conf is setup to route that vhost.

### CODE REVIEW

File: accounts/accounts/routes/ui.py [hunk 1]
```diff
@@ -192,3 +192,9 @@ def captcha() -> Response:
     token = request.args.get('token')
     data, code, headers = captcha_image.get(token, secret, request.remote_addr, font)
     return send_file(data['image'], mimetype=data['mimetype']), code, headers
+
+
+@blueprint.route('/auth_status', methods=['GET'])
+def auth_status() -> Response:
+    # TODO Test resurces like DB and redis
+    return make_response("OK")
```
Comment (DavidLFielding) [inline accounts/accounts/routes/ui.py:200]: Is there a ticket for this future TODO work?

File: accounts/accounts/factory.py [hunk 2]
```diff
@@ -20,6 +20,25 @@ def create_web_app() -> Flask:
...
+    # SERVER_NAME.
+
+    # The flask docs say: SERVER_NAME: the name and port number of the
+    # server. Required for subdomain support (e.g.: 'myapp.dev:5000')
+    # Note that localhost does not support subdomains so setting this
+    # to “localhost” does not help. Setting a SERVER_NAME also by
+    # default enables URL generation without a request context but with
+    # an application context.
+    app.config['SERVER_NAME'] = None
+
```
Comment (DavidLFielding) [inline accounts/accounts/factory.py:41]: Nice detailed explanation.

**Review by DavidLFielding — COMMENTED** [review]
Looks good!

File: accounts/accounts/routes/ui.py (hunk 1 above)
Comment (bdc34) [reply accounts/accounts/routes/ui.py:200]: Yes, I've found it nice to check not just that the service is running but also that any resources the service is dependent on are available and working.

**Pull request merged**